### PR TITLE
[Feature] Optimize catalog controller

### PR DESCRIPTION
**Pull request by hunter-cloud09**
re #70

### CODE REVIEW

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java [hunk 1]
```diff
@@ -18,24 +18,17 @@
...
-import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.DeleteMapping;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:31]: Do not use `*` to import the classes of the same package.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java [hunk 2]
```diff
@@ -99,14 +70,14 @@ public R<List<CatalogInfo>> getCatalog() {
...
+     * @param catalogId The catalog id.
      * @return A response indicating the success or failure of the operation.
      */
-    @DeleteMapping("/remove/{catalogName}")
-    public R<Void> removeCatalog(@PathVariable String catalogName) {
-        QueryWrapper<CatalogInfo> queryWrapper = new QueryWrapper<>();
-        queryWrapper.eq("catalog_name", catalogName);
-        return catalogService.remove(queryWrapper)
+    @DeleteMapping("/remove/{catalogId}")
+    public R<Void> removeCatalog(@PathVariable String catalogId) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:77]: Could this support remove by catalog name?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.data.dto;
+
+import lombok.Data;
+
+@Data
+public class CatalogDto {
+    private String catalogType;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java:25]: Does the DTO need catalog id?

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@hunter-cloud09, thanks for the contribution. I left some comments for this pull request. PTAL.
BTW, could these changes add some tests?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java (hunk 2 above)
Comment (hunter-cloud09) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:77]: Why use catalog name? Wouldn't it be better to use catalog id?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java (hunk 3 above)
Comment (hunter-cloud09) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java:25]: It's only a DTO. need't id.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java (hunk 2 above)
Comment (SteNicholas) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:83]: @hunter-cloud09, does the `removeCatalog` only support by id? IMO, the operation that removes catalog could support by catalog id or name.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java (hunk 3 above)
Comment (SteNicholas) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java:25]: @hunter-cloud09, why not DTO need id? Any standard?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java [hunk 4]
```diff
@@ -50,35 +49,13 @@ public class CatalogController {
...
+     * @param catalogDto The catalogDto for the catalog.
      * @return The created catalog.
      */
     @PostMapping("/create")
-    public R<Void> createCatalog(@RequestBody CatalogInfo catalogInfo) {
-        if (!catalogService.checkCatalogNameUnique(catalogInfo)) {
-            return R.failed(Status.CATALOG_NAME_IS_EXIST, catalogInfo.getCatalogName());
-        }
-
+    public R<Void> createCatalog(@RequestBody CatalogDto catalogDto) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:56]: Could you update other interfaces with DTO? like `createDatabase`?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java (hunk 2 above)
Comment (hunter-cloud09) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:83]: Do you mean to add an api?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java (hunk 3 above)
Comment (hunter-cloud09) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java:25]: The id is automatically generated by the snowflake model, so there is no need to pass in the id when creating the catalog.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java (hunk 2 above)
Comment (SteNicholas) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:83]: @hunter-cloud09, I mean that the parameter type could be `CatalogDTO`.

Comment (hunter-cloud09) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:83]: > @hunter-cloud09, I mean that the parameter type could be `CatalogDTO`.

get it!

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java [hunk 5]
```diff
@@ -97,17 +76,27 @@ public R<List<CatalogInfo>> getCatalog() {
...
      */
-    @DeleteMapping("/remove/{catalogName}")
-    public R<Void> removeCatalog(@PathVariable String catalogName) {
-        QueryWrapper<CatalogInfo> queryWrapper = new QueryWrapper<>();
-        queryWrapper.eq("catalog_name", catalogName);
-        return catalogService.remove(queryWrapper)
-                ? R.succeed()
-                : R.failed(Status.CATALOG_REMOVE_ERROR);
+    @PostMapping("/remove")
+    public R<Void> removeCatalog(@RequestBody RemoveCatalogDto removeCatalogDto) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/CatalogController.java:85]: Could this reuse `CatalogDto`?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/DatabaseController.java [hunk 6]
```diff
@@ -104,18 +113,19 @@ public R<List<DatabaseInfo>> getAllDatabases() {
...
+     * @param databaseDto The drop database DTO.
      * @return A response indicating the success or failure of the removal operation.
-     * @throws RuntimeException if the database is not found or it is not empty.
+     * @throws RuntimeException if the database is not found, or it is not empty.
      */
-    @DeleteMapping("/drop/{databaseName}/{catalogName}")
-    public R<Void> dropDatabase(
-            @PathVariable String databaseName, @PathVariable String catalogName) {
+    @PostMapping("/drop")
+    public R<Void> dropDatabase(@RequestBody DropDatabaseDto databaseDto) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/DatabaseController.java:121]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import lombok.Data;
+
+@Data
+public class CatalogDto {
+
+    private Integer id;
+
+    private String catalogType;
+
+    private String catalogName;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java:31]:
```suggestion
    private String name;
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+import com.baomidou.mybatisplus.annotation.TableLogic;
+import lombok.Data;
+
+@Data
+public class CatalogDto {
+
+    private Integer id;
+
+    private String catalogType;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CatalogDto.java:29]:
```suggestion
    private String type;
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CreateDatabaseDto.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.data.dto;
+
+import lombok.Getter;
+
+@Getter
+public class CreateDatabaseDto extends DatabaseDto {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/CreateDatabaseDto.java:24]: Only keep `DatabaseDto`.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/DatabaseDto.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package org.apache.paimon.web.server.data.dto;
+
+import lombok.Data;
+
+@Data
+public class DatabaseDto {
+    private String databaseName;
+    private String catalogId;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/DatabaseDto.java:26]: Add new line between fields?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/DropDatabaseDto.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.data.dto;
+
+import lombok.Getter;
+
+@Getter
+public class DropDatabaseDto extends DatabaseDto {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/DropDatabaseDto.java:24]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/RemoveCatalogDto.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.server.data.dto;
+
+import lombok.Data;
+
+@Data
+public class RemoveCatalogDto {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/RemoveCatalogDto.java:24]: Ditto.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/CatalogService.java [hunk 13]
```diff
@@ -31,5 +33,13 @@ public interface CatalogService extends IService<CatalogInfo> {
      * @param catalog catalog info
      * @return result
      */
-    boolean checkCatalogNameUnique(CatalogInfo catalog);
+    boolean checkCatalogNameUnique(CatalogDto catalog);
+
+    /**
+     * Create a catalog.
+     *
+     * @param catalogDto catalogInfo for the catalog.
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/CatalogService.java:41]:
```suggestion
     * @param catalogDto catalog for the catalog.
```

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@hunter-cloud09, thanks for updates. I left some comments for this update. BTW, could all dtos name like `TableDTO`? PTAL.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/LoginController.java [hunk 14]
```diff
@@ -48,7 +48,7 @@ public class LoginController {
      * @return token string
      */
     @PostMapping("/login")
-    public R<UserInfoVo> login(@RequestBody LoginDto loginDto) {
+    public R<UserInfoVo> login(@RequestBody LoginDTO loginDto) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/LoginController.java:51]:
```suggestion
    public R<UserInfoVo> login(@RequestBody LoginDTO loginDTO) {
```

Comment (SteNicholas) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/LoginController.java:51]: Also update the comment of `loginDTO`.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/DatabaseDTO.java [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+package org.apache.paimon.web.server.data.dto;
+
+import lombok.Data;
+
+@Data
+public class DatabaseDTO {
+
+    private String databaseName;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/DatabaseDTO.java:26]:
```suggestion
    private String name;
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/LoginDTO.java [hunk 16]
```diff
@@ -24,7 +24,7 @@
 
 /** login dto. */
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/LoginDTO.java:25]:
```suggestion
/** The DTO of login. */
```

Comment (SteNicholas) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/data/dto/LoginDTO.java:25]: Other DTOs also add the comment of class.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/UserService.java [hunk 17]
```diff
@@ -36,7 +36,7 @@ public interface UserService extends IService<User> {
      * @param loginDto login params
      * @return {@link String}
      */
-    UserInfoVo login(LoginDto loginDto) throws BaseException;
+    UserInfoVo login(LoginDTO loginDto) throws BaseException;
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/UserService.java:39]:
```suggestion
    UserInfoVo login(LoginDTO loginDTO) throws BaseException;
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/UserServiceImpl.java [hunk 18]
```diff
@@ -69,7 +69,7 @@ public class UserServiceImpl extends ServiceImpl<UserMapper, User> implements Us
      * @return {@link String}
      */
     @Override
-    public UserInfoVo login(LoginDto loginDto) throws BaseException {
+    public UserInfoVo login(LoginDTO loginDto) throws BaseException {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/UserServiceImpl.java:72]:
```suggestion
    public UserInfoVo login(LoginDTO loginDTO) throws BaseException {
```

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@hunter-cloud09, thanks for updates. I left some minor comments. PTAL.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java [hunk 19]
```diff
@@ -68,18 +68,18 @@ public class TableController {
     /**
      * Creates a table in the database based on the provided TableInfo.
      *
-     * @param tableInfo The TableInfo object containing information about the table.
+     * @param tableDTO The TableInfo object containing information about the table.
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/TableController.java:71]:
```suggestion
     * @param tableDTO The TableDTO object containing information about the table.
```

**Review by SteNicholas — APPROVED** [review]
LGTM. Thanks for @hunter-cloud09 updates.

**Pull request merged**